[Mellanox]Adding SKU Mellanox-SN4700-O32 and Mellanox-SN4700-V64 | internal | master

ayurkiv-nvda opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it
A new SKUs for MSN4700 Platform: **Mellanox-SN4700-O32** and **Mellanox-SN4700-V64** 

Requirements for **Mellanox-SN4700-O32**:

- **8 x 400Gbps** uplink to T2 switch (O13 to O20)
- **24 x 400Gbps** downlinks to T0 switch (O1-O12, O21-O32)
- **Breakout mode** No breakout mode. All ports working in 400Gb mode. .
- **Auto-negotiation** : Yes
- **FEC mode:** RS 
- **Type of transceiver**: 400Gb Optical.
- **warm boot should be supported**  “No for T1 role”
- **VxLAN source port range set** N/A
- **Static Policy Based Hashing supported** N/A
- **Cable length** “T0-T1 40m; T1-T2 2000m”
- **Tradition buffer model is must** “Yes”
- **Shared headroom should be supported** “Yes”
- **Over-subscription ratio:** “2”.

Requirements for **Mellanox-SN4700-V64**
 - **16 x 200Gbps** uplink to T1 switch (V-25&V26 to V-39&40)
 - **48 x 200Gbps** downlinks to servers (Left panel downlink ports: V-1&2 to V-23&24; Right panel downlink ports: V-41&42 to V-63&64)
- **Breakout mode** split from 400Gbps ports (2x200)
- **Auto-negotiation** : Yes
- **FEC mode:** RS 
- **Type of transceiver**: 200Gb AOC between T0 and T1; 200Gb DAC between T0 and host.
- **warm boot should be supported**  “Yes for T0 role”
- **VxLAN source port range set** N/A
- **Static Policy Based Hashing supported** N/A
- **Cable length** “T0-T1 40m, T0-Server 5m”
- **Tradition buffer model is must** “Yes”
- **Shared headroom should be supported** “Yes”
- **Over-subscription ratio:** “2”.

**Additional Details:**
- QoS configs for Mellanox-SN4700-V64 updated in order to fulfill Dual-ToR  buffer (+DSCP remapping) requirements


##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

#### How to verify it

<!--
If PR needs to be backported, then the PR must be tested against the base branch and the earliest backport release branch and provide tested image version on these two branches. For example, if the PR is requested for master, 202211 and 202012, then the requester needs to provide test results on master and 202012.
-->

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

<!--
- Please provide tested image version
- e.g.
- [x] 20201231.100
-->

- [ ] <!-- image version 1 -->
- [ ] <!-- image version 2 -->

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

<!--
 Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.
-->

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

stephenxs commented:
in the description `48 x 400Gbps downlinks to servers (Left panel downlink ports: V-1&2 to V-23&24; Right panel downlink ports: V-41&42 to V-63&64)` typo? 48 * 200G downlinks?

ayurkiv-nvda commented:
> @ayurkiv-nvda have you also updated the sonic utilities repo? https://github.com/sonic-net/sonic-utilities/blob/b6f7c2b7d138299475d994d251721455deab668f/generic_config_updater/gcu_field_operation_validators.conf.json#L23

Added for master branch
for 202211 there are no such file at alll
https://github.com/ayurkiv-nvda/sonic-utilities/pull/24
